### PR TITLE
Fix permission bugs in team API

### PR DESCRIPTION
**Pull request by ethantkoenig**
Some endpoints unnecessarily required the authenticated user to be an admin (basically everything removed from `api/v1/admin/org_team.go`).

Other endpoints didn't check that the authenticated user was an organization member.

### CODE REVIEW

File: routers/api/v1/org/team.go [hunk 1]
```diff
@@ -10,11 +10,16 @@ import (
 	"code.gitea.io/gitea/models"
 	"code.gitea.io/gitea/modules/context"
 	"code.gitea.io/gitea/routers/api/v1/convert"
+	"code.gitea.io/gitea/routers/api/v1/user"
 )
 
 // ListTeams list all the teams of an organization
 func ListTeams(ctx *context.APIContext) {
 	org := ctx.Org.Organization
+	if !org.IsOrgMember(ctx.User.ID) {
```
Comment (lunny) [inline routers/api/v1/org/team.go:19]: This depends public or private status of org members. If public, then everyone could visit the members.

File: routers/api/v1/org/team.go [hunk 2]
```diff
@@ -63,3 +62,112 @@ func GetTeamRepos(ctx *context.APIContext) {
...
+	if err := models.DeleteTeam(ctx.Org.Team); err != nil {
+		ctx.Error(500, "DeleteTeam", err)
+		return
+	}
+	ctx.Status(204)
+}
+
+// GetTeamMembers api for get a team's members
+func GetTeamMembers(ctx *context.APIContext) {
+	if !models.IsOrganizationMember(ctx.Org.Team.OrgID, ctx.User.ID) {
```
Comment (lunny) [inline routers/api/v1/org/team.go:124]: ^

**Review by lunny — CHANGES_REQUESTED** [review]

File: routers/api/v1/org/team.go (hunk 1 above)
Comment (ethantkoenig) [reply routers/api/v1/org/team.go:19]: I don't understand what you are saying; this is an endpoint for viewing an organization's teams, not for viewing the members of those teams.

Regardless, the corresponding endpoint in the Github API (`GET /orgs/:orgname/teams`) is only accessible to organization members, so I believe that is what we want to do here

File: routers/api/v1/org/team.go (hunk 2 above)
Comment (ethantkoenig) [reply routers/api/v1/org/team.go:124]: The corresponding Github endpoint (`GET /teams/:id`) is only accessible to organization members (even if there are public members of the team), so I believe that is what we want to do here.

**Review by lunny — APPROVED** [review]

**Comment by lunny** [comment]
LGTM

**Comment by Bwko** [comment]
LGTM

**Comment by lunny** [comment]
conflicted.

**Review by appleboy — APPROVED** [review]

**Comment by appleboy** [comment]
LGTM

**Comment by ethantkoenig** [comment]
~~Don't merge yet, there was something weird resolving the conflict~~

Fixed

**Comment by thomsh** [comment]
Hello,
@ethantkoenig with this old commit you remove two use full endpoints : 
```
/admin/orgs/:orgname
/admin/teams/:teamid
```
Actually, in v1.1.2, there is no API endpoint to replace them.

If you didn't remove anything in the` /admin/users` endpoint, 
there was no permission bug with orgs and teams endpoints ? Or not?
_I'm just trying to understand ;)_

Maybe I should open a issue about that ?

Have a nice day

**Comment by ethantkoenig** [comment]
@theznx The `POST /admin/orgs/:orgname/` endpoint was moved to `POST /orgs/:orgname`, and the `PATCH /admin/teams/:teamid` and `DELETE /admin/teams/:teamid` endpoints were moved to `PATCH /teams/:teamid` and `DELETE /teams/:teamid` respectively.

**Comment by thomsh** [comment]
@ethantkoenig Oh! Sorry for the inconvenience.

**Pull request merged**